### PR TITLE
Add layout and products api

### PR DESCRIPTION
**Pull request by priyankashrma**

### CODE REVIEW

**Comment by priyankashrma** [comment]
image

File: src/store/user/actions.js [hunk 1]
```diff
@@ -14,8 +14,10 @@ export function signInAction({ commit }, payload) {
   firebase
     .auth()
     .signInWithEmailAndPassword(payload.email, payload.password)
-    .then(() => {
-      this.$router.push("/");
+    .then(s => {
```
Comment (santhoshgl) [inline src/store/user/actions.js]: @priyankashrma give a proper name here instead of just **s**

File: src/pages/Dashboard.vue [hunk 2]
```diff
@@ -1,29 +1,28 @@
...
 
 <script>
 import firebase from "firebase";
+import { mapGetters, mapActions } from "vuex";
 
 export default {
   name: "LayoutDefault",
   computed: {
-    ...mapGetters(["getUser", "isUserAuth"])
+    ...mapGetters("user", ["getUser", "isUserAuth"])
```
Comment (santhoshgl) [inline src/pages/Dashboard.vue]: @priyankashrma you can get rid off passing "user" every time like this, by removing the **namespaced:true** over here https://github.com/priyankashrma/quasar-login-signup/blob/master/src/store/user/index.js#L7

File: src/pages/Login.vue [hunk 3]
```diff
@@ -87,16 +87,20 @@ export default {
     };
   },
   computed: {
-    ...mapGetters(["isUserAuth"])
+    ...mapGetters("user", ["isUserAuth"])
```
Comment (santhoshgl) [inline src/pages/Login.vue]: @priyankashrma same here as well.

File: src/pages/Login.vue [hunk 4]
```diff
@@ -87,16 +87,20 @@ export default {
...
   },
   methods: {
-    ...mapActions(["signInAction"]),
+    ...mapActions("user", ["signInAction"]),
```
Comment (santhoshgl) [inline src/pages/Login.vue]: @priyankashrma same here as well.

File: src/pages/Products.vue [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+</template>
+
+<script>
+import Product from "../components/Product.vue";
+
+export default {
+  data() {
+    return { products: [] };
+  },
+  beforeCreate() {
```
Comment (santhoshgl) [inline src/pages/Products.vue]: @priyankashrma move this to store create a new store called product and place inside action.js file, we should not call the apis inside the .vue files.

File: src/pages/SignUp.vue [hunk 6]
```diff
@@ -123,39 +122,24 @@ export default {
       password: "",
       phone: "",
       firstname: "",
-      lastname: ""
+      lastname: "",
+      isPwd: true
     };
   },
+  computed: {
+    ...mapGetters("user", ["isUserAuth"])
```
Comment (santhoshgl) [inline src/pages/SignUp.vue]: @priyankashrma same here.

File: src/pages/SignUp.vue [hunk 7]
```diff
@@ -123,39 +122,24 @@ export default {
...
+  },
   methods: {
+    ...mapActions("user", ["signUpAction"]),
```
Comment (santhoshgl) [inline src/pages/SignUp.vue]: @priyankashrma same here

File: src/pages/Products.vue (hunk 5 above)
Comment (priyankashrma) [reply src/pages/Products.vue]: I have made the required changes, but getProductsAction can't be called. Please check the screenshot.

image

Comment (santhoshgl) [reply src/pages/Products.vue]: @priyankashrma please call the action inside created hook

**Review by santhoshgl — APPROVED** [review]

**Pull request merged**